scripts/page_browse.js

```diff
@@ -0,0 +1,247 @@
+};
+
+// Construct the buttons for horizontal/vertical split
+// and return a splitter variable.
+var viewSplitter = function(container) {
+    const textImageSplitID = "text_image_split";
+    const splitPercentID = "split_percent";
+    let splitter = splitControl();
+
+    // stored value is "horizontal" or "vertical"
+    let splitDirString = localStorage.getItem(textImageSplitID);
+    if(!splitDirString) {
+        splitDirString = "horizontal";
+    }
+    // splitDirection is a value defined in splitter
+    let splitDirection = (splitDirString === "horizontal") ? splitter.DIRECTION.HORIZONTAL : splitter.DIRECTION.VERTICAL;
+
+    let splitPercent = localStorage.getItem(splitPercentID);
+    if(!splitPercent) {
+        splitPercent = 50;
+    }
+
+    let mainSplit = splitter.setup(container, {splitDirection: splitDirection, splitPercent: splitPercent});
+
+    // must set the image width, otherwise it will at first be small, then when image loads later elements can wrap and disappear
+    let vSwitchButton = $("<button>", {type: 'button', class: 'img-button', title: dpData.strings.switchVert}).append($("<img>", {src: dpData.buttonImages.imgVSplit, width: 27}));
+    let hSwitchButton = $("<button>", {type: 'button', class: 'img-button', title: dpData.strings.switchHoriz}).append($("<img>", {src: dpData.buttonImages.imgHSplit, width: 27}));
+
+    function setSplitControls(splitDirection) {
+        if (splitDirection === splitter.DIRECTION.VERTICAL) {
+            hSwitchButton.show();
+            vSwitchButton.hide();
+        } else {
+            hSwitchButton.hide();
+            vSwitchButton.show();
+        }
+    }
+
+    function changeSplit(splitDirection) {
+        mainSplit.setSplit(splitDirection);
+        setSplitControls(splitDirection);
+        splitDirString = (splitDirection === splitter.DIRECTION.HORIZONTAL) ? "horizontal" : "vertical";
+        localStorage.setItem(textImageSplitID, splitDirString);
+    }
+
+    vSwitchButton.click(function () {
+        changeSplit(splitter.DIRECTION.VERTICAL);
+    });
+
+    hSwitchButton.click(function () {
+        changeSplit(splitter.DIRECTION.HORIZONTAL);
+    });
+
+    setSplitControls(splitDirection);
+
+    mainSplit.dragEnd.add(function (percent) {
+        localStorage.setItem(splitPercentID, percent);
+    });
+
+    return {
+        mainSplit: mainSplit,
+        buttons: [vSwitchButton, hSwitchButton],
+    };
+};
+
+// Construct the image sizing controls.
+var imageControl = function() {
+    const imagePercentID = "image_percent";
+    let percent = localStorage.getItem(imagePercentID);
+    if(!percent) {
+        percent = 100;
+    }
+
+    let percentInput = $("<input>", {type: 'number', min: '1', max: '999', value: percent});
+    let resizeButton = $("<input>", {type: 'button', value: dpData.strings.resize});
+
+    let image = $("<img>", {src: dpData.projectsUrl + "/" + pageBrowserData.projectid + "/" + pageBrowserData.currentPage});
```

Comment:
We should include the `$project->url` in the `pageBrowserData` object and not try to reconstruct it here from the `projectsUrl`.